### PR TITLE
Cloudtest improvements

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin <denis.tingajkin@xored.com>

## Changes:
1) Added native artifacts support. I.e. each test has its own folder for artifacts.
2) Added a possibility to import all executions from the folder.
3) Added scripts before/after for executions.
4) Added possible to assign go tasks by test names.
5) Fixed bug with removing rescheduled test artifacts.
6) Changed tests shuffling. 
7) Added unit tests to cover.

### CODE REVIEW

File: pkg/commands/cloud_config_import.go [hunk 1]
```diff
@@ -170,11 +172,35 @@ func CloudTestRun(cmd *cloudTestCmd) {
 
 func performImport(testConfig *config.CloudTestConfig) error {
 	for _, imp := range testConfig.Imports {
+		if strings.HasSuffix(imp, "*") {
```
Comment (haiodo) [inline pkg/commands/cloud_config_import.go]: May be better to have a full scope wildcards? replace all with .* and match as regular expression?

File: pkg/commands/cloud_config_import.go [hunk 2]
```diff
@@ -845,6 +871,36 @@ func (ctx *executionContext) executeTask(task *testTask, clusterConfigs []string
...
 		for _, inst := range instances {
 			inst.taskCancel = cancel
+			if task != nil && task.test != nil && inst.runningExecution != task.test.ExecutionConfig {
+				if inst.runningExecution != nil {
+					for _, cfg := range clusterConfigs {
+						err = ctx.handleScript(&runScriptArgs{
+							Name:          "After",
+							ClusterTaskId: task.clusterTaskID,
+							Script:        inst.runningExecution.After,
+							Env:           append(inst.runningExecution.Env, fmt.Sprintf("KUBECONFIG=%v", cfg)),
```
Comment (haiodo) [inline pkg/commands/cloud_config_import.go]: I think we need to pass Cloud config variable as parameter, to less depend on kubernetes.

File: pkg/commands/cloud_config_import.go [hunk 3]
```diff
@@ -861,10 +917,17 @@ func (ctx *executionContext) executeTask(task *testTask, clusterConfigs []string
 				_, _ = writer.WriteString(msg + "\n")
 				_ = writer.Flush()
 
-				onFailErr := ctx.handleOnFailTask(task, []string{fmt.Sprintf("KUBECONFIG=%s", cfg)}, writer)
+				onFailErr := ctx.handleScript(&runScriptArgs{
+					Name:          "OnFail",
+					ClusterTaskId: task.clusterTaskID,
+					Script:        task.test.ExecutionConfig.OnFail,
+					Env:           append(task.test.ExecutionConfig.Env, fmt.Sprintf("KUBECONFIG=%v", cfg)),
```
Comment (haiodo) [inline pkg/commands/cloud_config_import.go]: Same here.

File: pkg/commands/cloud_config_import.go (hunk 2 above)
Comment (denis-tingajkin) [reply pkg/commands/cloud_config_import.go]: Can we solve this problem within issue https://github.com/networkservicemesh/cloudtest/issues/1 ?

**Comment by denis-tingajkin** [comment]
@haiodo CI result for NSM repo https://github.com/networkservicemesh/networkservicemesh/pull/2057

File: pkg/commands/cloud_config_import.go (hunk 2 above)
Comment (haiodo) [reply pkg/commands/cloud_config_import.go]: ok

**Pull request merged**